service/raft/group0_state_machine.hh

```diff
@@ -94,6 +77,12 @@ struct group0_command {
 // Raft state machine implementation for managing group 0 changes (e.g. schema changes).
 // NOTE: group 0 raft server is always instantiated on shard 0.
 class group0_state_machine : public raft_state_machine {
+    enum class reload_module {
+        service_levels_cache
+    };
+    using modules_to_reload = enum_set<super_enum<reload_module,
+        reload_module::service_levels_cache>>;
```

Comment:
Why use `enum_set` here? Isn't the set known only in runtime?

Reply:
`enum_set` is not a compile-time set, it's just a dynamic set of bits based on the passed enum. It requires passing not a regular C++ enum but rather a `super_enum` which - for some reason - requires you to specify all possible variants you want to use in it as template parameters.

Reply:
The comment in the code says:

```
/**
 *
 * Allows to take full advantage of compile-time information when operating
 * on a set of enum values.
 *      
 * Examples:
 *      
 *   enum class x { A, B, C };
 *   using my_enum = super_enum<x, x::A, x::B, x::C>;
 *   using my_enumset = enum_set<my_enum>; 
 *  
 *   static_assert(my_enumset::frozen<x::A, x::B>::contains<x::A>(), "it should...");
 *
 *   assert(my_enumset::frozen<x::A, x::B>::contains(my_enumset::prepare<x::A>()));
 *
 *   assert(my_enumset::frozen<x::A, x::B>::contains(x::A));
 *
 */

```

We do not need any of that and I find its usage to be just a code obfuscation to be honest.

Reply:
I suggested using it because previously we were using `std::unordered_set`, which seems quite heavy for the purpose of just keeping one or two integers. I don't think we have a less clunky abstraction which could let us implement a type-safe bitset.

Maybe we could turn this into a struct of bools?

```c++
struct modules_to_reload {
    bool service_levels_cache = false;
};
```

Reply:
Yes, `std::unordered_set` is overkill. The struct is straightforward IMO.

Reply:
It's a layering violation, no? It make group0 know about all its clients, but all its clients have to known about group0.

Better to register callbacks table_id -> callback. I don't think an unordered_set is a problem. The goal here is to disentangle the code, not save a few cycles in an administrative function.

Reply:
Changed to use struct of bools instead of enum_set